pkg/kubelet/kubelet_network_linux.go

```diff
@@ -95,27 +97,37 @@ func (kl *Kubelet) syncNetworkUtil(iptClient utiliptables.Interface) bool {
 		return false
 	}
 
+	if _, err := iptClient.EnsureChain(utiliptables.TableFilter, KubeFirewallInChain); err != nil {
+		klog.Errorf("Failed to ensure that %s chain %s exists: %v", utiliptables.TableFilter, KubeFirewallInChain, err)
+		return false
+	}
+
 	// drop all non-local packets to localhost if they're not part of an existing
 	// forwarded connection. See #90259
 	if !iptClient.IsIPv6() { // ipv6 doesn't have this issue
-		if _, err := iptClient.EnsureRule(utiliptables.Append, utiliptables.TableFilter, KubeFirewallChain,
+		if _, err := iptClient.EnsureRule(utiliptables.Append, utiliptables.TableFilter, KubeFirewallInChain,
 			"-m", "comment", "--comment", "block incoming localnet connections",
 			"--dst", "127.0.0.0/8",
-			"!", "--src", "127.0.0.0/8",
+			"!", "-i", "lo",
 			"-m", "conntrack",
 			"!", "--ctstate", "RELATED,ESTABLISHED,DNAT",
```

Comment:
Do we still need the ctstate match ?

Reply:
(still before coffee) yes

Reply:
I believe so, yes. imagine you have

```
ipables -t nat -A PREROUTING -m tcp --dst-port 443 -j DNAT --to-destination 127.0.0.1:6443
```

while this is slightly convoluted, we should definitely allow this to pass through.